app/models/web_hook/event.rb

```diff
@@ -137,11 +137,14 @@ def push_event?
 
     delegate :connection, to: 'ActiveRecord::Base'
 
+    # :reek:NilCheck
+    # That is fine as we want really to check for nil value in the changes
     def guess_event
+      resource_changes = @resource.previous_changes
       case
       when @resource.destroyed?
         'deleted'
-      when @resource.created_at.present? && @resource.created_at == @resource.updated_at
+      when resource_changes.key?('created_at') && resource_changes['created_at'].first.nil?
```

Comment:
Maybe you could do `when (created_at_changes = @resource.previous_changes['created_at']) && created_at_changes.first.nil?` and then you don't need the previous `resource_changes = @resource.previous_changes`

Reply:
Why? I don't see any complain for this in `rubocop` or `reek` (line `146` doing it the way I wrote here):

![image](https://user-images.githubusercontent.com/11318903/51681968-807b1580-1fe6-11e9-9a1c-b3e1e5f0168c.png)

![image](https://user-images.githubusercontent.com/11318903/51681915-5c1f3900-1fe6-11e9-92e4-cb8aa6ca663f.png)

![image](https://user-images.githubusercontent.com/11318903/51681931-64777400-1fe6-11e9-8bd2-748446849677.png)

Reply:
Anyway that won't work. I need to check if the key `created_at` is present.

Reply:
It works because if it is not present, it returns `nil` 😄 So `(created_at_changes = @resource.previous_changes['created_at'])` would return `nil` which is a falsy value and wouldn't check the next line 😄

Reply:
👍 But I do not know. I like the way we do not assign variables in a conditional.
Strange that rubocop does not complain :/ https://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Lint/AssignmentInCondition

It only takes the `if` or `unless` and not `when`

Reply:
It does not complain also for the brakets, look:
![image](https://user-images.githubusercontent.com/11318903/51686432-a7d6e000-1ff0-11e9-8af1-dd6985f3f41c.png)
![image](https://user-images.githubusercontent.com/11318903/51686523-d05eda00-1ff0-11e9-9632-91b68ffbe3a8.png)

Look that in the link you gave me, it is comparing between `=` and `==` because the problem for Rubocop is that it is easy to confuse them both 😄 

![image](https://user-images.githubusercontent.com/11318903/51686687-1fa50a80-1ff1-11e9-8b0a-c9ddeba553e8.png)

Btw, I personally like more the `if` format for this than the empty `case` 😄 But all this is all just about what we like more, not the end of the world 😄